PatchesAndAreasApi/V1/Gateways/PatchesGateway.cs

```diff
@@ -95,6 +95,23 @@ public async Task<PatchesDb> DeleteResponsibilityFromPatch(DeleteResponsibilityF
 
             return patch;
         }
+
+        public async Task<PatchesDb> ReplacePatchResponsibleEntities(PatchesQueryObject query, List<ResponsibleEntities> responsibleEntitiesRequestObject,
+                                                                      int? ifMatch)
+        {
+            _logger.LogDebug($"Calling IDynamoDBContext.LoadAsync for id {query.Id} and then IDynamoDBContext.SaveAsync");
+            var patch = await _dynamoDbContext.LoadAsync<PatchesDb>(query.Id).ConfigureAwait(false);
+            if (patch == null) return null;
+            if (ifMatch != patch.VersionNumber)
+                throw new VersionNumberConflictException(ifMatch, patch.VersionNumber);
+
+            //update responsibleEntity with request sent
+            patch.ResponsibleEntities = responsibleEntitiesRequestObject;
+
+            await _dynamoDbContext.SaveAsync(patch).ConfigureAwait(false);
+
+            return patch;
+        }
```

Comment:
I may have missed some context here. Should this not be a PUT endpoint? The endpoint is pointing to `/responsible-entities` as a resource, where the whole object is replaced. 

From my understanding, a patch endpoint is to partially update the entity with optional fields (as we have done in other APIS). As opposed to PUT, where you pass a replacement for the entity.

Reply:
I guess we were thinking of it as a PATCH of the Patch resource rather than a PUT of the "Responsible Entities for a Patch", but especially since it's just a "take and replace" operation, PUT may be more intuitive.